packages/headless/src/controllers/core/facets/facet/headless-core-facet.ts

```diff
@@ -464,8 +464,17 @@ export function buildCoreFacet(
       const response = getResponse();
       const isLoading = getIsLoading();
       const enabled = getIsEnabled();
+      let sortCriterion!: FacetSortCriterion;
+
+      if (typeof request.sortCriteria === 'object') {
+        sortCriterion =
+          request.sortCriteria.order === 'descending'
+            ? 'alphanumericDescending'
+            : 'alphanumeric';
+      } else {
+        sortCriterion = request.sortCriteria;
+      }
```

Comment:
IMO this logic should instead be part of the `facet-set-slice`, since it already contains a `request` object meant to be a match with what we'll send to the API.

Reply:
what if we decided not to support the `alphanumericDescending` string at all (both in the slice and the controller), and required it to be an object? Would that be possible?

While changing the type in `FacetRequest` to accept both the string or an object would be breaking, I don't think the cases where it's actually breaking would be fairly narrow (e.g., somebody casts it to `string` or does some string operations on it). Given that this is a feature and not a fix, I think we also have a bit more leeway to do something potentially (but unlikely) breaking.

Reply:
After some discussion over slack, we have decided to continue with the current approach to save headache's from supporting multiple types at once on both our/my end as well as the implementers end. We can support this model end-to-end as a part of V3 once it's been more robust on the Search API's ✅